### ISSUE TITLE
Wagmi does not work on next.js  `_app.js` file anymore

### ISSUE THREAD

**Issue by hskang9** [wevm/wagmi#3488]
### Describe the bug

I am using next.js 13 with wagmi supported by Vercel, the creator of Next.js. previous version of wagmi supported using `useAccount` hook, but current version don't support that and the team did not even add a breaking changes. Now the team only provides typescript playground or wagmi template with Vite boilerplate. I really don't understand why these breaking changes were not even specified and try to make Next.js devs reproduce examples hard. Vite is not only supporting vercel's api functions, and it is not aimed for building fullstack applications. Bring back the feature which wagmi supported before asap. 

### Link to Minimal Reproducible Example

[https://github.com/hskang9/nextjs-support-gone](https://github.com/hskang9/nextjs-support-gone)

### Steps To Reproduce

The process is easy to follow unless you haven't coded anything.
1. generate next.js app example with the command `npx create-wc-dapp@latest -y w3m`.
2. In _app.js, import useAccount from wagmi.
3. add hook `const { isConnected } = useAccount();` inside App component.
4. run the app with command `yarn dev`.

### Wagmi Version

1.4.2

### Viem Version

1.21.4

### TypeScript Version

5.3.3

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Anything else?

You self-proclaim that you made the new so-called "reactivity" for Ethereum. While it is supposed to provide predictable implementations, it is now more unpredictable than ever with breaking changes in every major version of Viem, unnecessary implementation of Typescript and huge bias on using this on Vite by excluding minimal producible example with Next.js(which is what this should support). You self-proclaim there are some breaking changes, but this is literally breaking every existing application developers' work without any full specification. This library should actually be named as "very opinionated api to connect to Ethereum with bunch of unpredictable changes which only Vite boilerplate can be safe to build". It is as if the team just play in solitary confinement, separated from real world whereas application developers suffer from chasing their breaking changes. They just count from 1 to 10 with new, "shiny" libraries to mix up and always forget after 3 seconds.

**Comment by hskang9** [comment]
As @tmm can't even follow instructions, I put repo link.

**Comment by tmm** [comment]
Took a look at your reproduction and it's an easy fix. You cannot use Wagmi Hooks outside of the `WagmiConfig`. For example, [here](https://github.com/hskang9/nextjs-support-gone/blob/main/src/pages/_app.tsx#L46) you use the `useAccount` hook outside of the `WagmiConfig`. This does not work as `useAccount` depends on `WagmiConfig` to inject `config` via React Context.

Separately, I want to reiterate minimal reproductions save us time and help us help you faster. [We've been over this before with you](https://github.com/wevm/wagmi/issues/2986#issuecomment-1712150093). If you continue to not follow the [Code of Conduct](https://github.com/wevm/wagmi#our-standards), you will get temporarily banned from interacting with our GitHub organization. Any subsequent violations after a temporary ban will lead to a permanent ban.

**Comment by hskang9** [comment]
version 0 worked well, and why do you not add that in v1 changes? This is lack of integrity and you are forcing application developers destroy their codes.

**Comment by jxom** [comment]
Using Hooks outside of the `WagmiConfig` has always been incorrect usage, and we have always guided users to use Hooks within its context. So it’s not really a breaking change tbh. The reason why it worked in v0 is because we relied on a singleton global config in Wagmi Core. Technically, we did note that breaking change right here: https://wagmi.sh/core/guides/migrate-from-v1-to-v2#removed-config-singleton
